extensions/amp-facebook-comments/0.1/amp-facebook-comments.js

```diff
@@ -69,11 +76,39 @@ class AmpFacebookComments extends AMP.BaseElement {
     listenFor(iframe, 'embed-size', data => {
       this./*OK*/changeHeight(data['height']);
     }, /* opt_is3P */true);
+    this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleFacebookMessages_.bind(this)
+    );
+    this.toggleLoading(true);
     this.element.appendChild(iframe);
     this.iframe_ = iframe;
     return this.loadPromise(iframe);
   }
 
+  /** @private */
+  handleFacebookMessages_(event) {
+    if (this.iframe_ && event.source != this.iframe_.contentWindow) {
+      return;
+    }
+    if (!getData(event) || !(isObject(getData(event))
+        || startsWith(/** @type {string} */ (getData(event)), '{'))) {
```

Comment:
Attempting to `parseJson` might be a better check of whether it is valid JSON rather than checking the first character. What do you think?

Reply:
Yes. Definitely. Didn't know a function like `tryParseJSon` existed. Changing the code to use that instead.